azure/apimanagement/main.tf

```diff
@@ -61,4 +63,68 @@ resource "azurerm_api_management_backend" "api" {
   api_management_name = azurerm_api_management.api.name
   protocol            = "http"
   url                 = "${var.api_backend_url}"
+}
+
+resource "azurerm_api_management_product" "api" {
+  product_id            = "product-${var.organization}-${var.project}"
+  api_management_name   = azurerm_api_management.api.name
+  resource_group_name   = azurerm_resource_group.api.name
+  display_name          = "Product ${var.organization} ${var.project}"
+  description           = "Test product with terraform"
+  subscription_required = true
+  published             = true
+}
+
+resource "azurerm_api_management_product_policy" "api" {
+  product_id          = azurerm_api_management_product.api.product_id
+  api_management_name = azurerm_api_management_product.api.api_management_name
+  resource_group_name = azurerm_api_management_product.api.resource_group_name
+
+  xml_content = var.policy_product
+
+}
+
+resource "azurerm_api_management_product_api" "api" {
+  api_name            = azurerm_api_management_api.api.name
+  product_id          = azurerm_api_management_product.api.product_id
+  api_management_name = azurerm_api_management.api.name
+  resource_group_name = azurerm_api_management.api.resource_group_name
+  
+}
+
+data "azurerm_api_management_group" "api" {
+  name                = "developers"
+  api_management_name = azurerm_api_management.api.name
+  resource_group_name = azurerm_api_management.api.resource_group_name
+}
+
+data "azurerm_api_management_group" "api2" {    
```

Comment:
Why are we duplicating the creation of the same resource type? If we can't interact through a list passed by the service itself, then we should rethink where this groups must be created, since we don't know from a service perspective how many they are.